### PR TITLE
Add 'encryptionKMSID' for 'Attach Storage' form

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
Follow-up of https://github.com/red-hat-storage/odf-console/pull/1662

https://issues.redhat.com/browse/DFBUGS-908

Screenshot 2024-11-26 at 8 13 32 PM

### CODE REVIEW

**Comment by alfonsomthd** [comment]
LGTM. @bipuladh @GowthamShanmugam @vbnrh to tag.

File: packages/odf/components/attach-storage-storagesystem/attach-storage-footer.tsx [hunk 1]
```diff
@@ -23,9 +23,17 @@ export const AttachStorageFormFooter = (
...
+  const { name, enableStorageClassEncryption, encryptionKMSID } =
+    storageClassDetails;
   const isDisabled =
-    checkRequiredValues(poolName, replicaSize, lsoStorageClassName, name) ||
+    checkRequiredValues(
+      poolName,
+      replicaSize,
+      lsoStorageClassName,
+      name,
+      enableStorageClassEncryption,
```
Comment (GowthamShanmugam) [inline packages/odf/components/attach-storage-storagesystem/attach-storage-footer.tsx:34]: Passing enableStorageClassEncryption checkRequiredValue, are we making storage encryption mandatory for attaching storage?

File: packages/odf/components/attach-storage-storagesystem/attach-storage-footer.tsx [hunk 2]
```diff
@@ -23,9 +23,17 @@ export const AttachStorageFormFooter = (
...
+      encryptionKMSID
```
Comment (GowthamShanmugam) [inline packages/odf/components/attach-storage-storagesystem/attach-storage-footer.tsx:35]: is just checking encryptionKMSID and enableStorageClassEncryption enough to decide whether the user provides all KMS fields?

File: packages/odf/components/attach-storage-storagesystem/attach-storage-footer.tsx (hunk 1 above)
Comment (SanjalKatiyar) [reply packages/odf/components/attach-storage-storagesystem/attach-storage-footer.tsx:34]: nope, not mandatory... hence the ternary condition...

File: packages/odf/components/attach-storage-storagesystem/attach-storage-footer.tsx (hunk 2 above)
Comment (SanjalKatiyar) [reply packages/odf/components/attach-storage-storagesystem/attach-storage-footer.tsx:35]: yes, it is same as what happens in StorageClass creation form currently...

**Review by GowthamShanmugam — APPROVED** [review]

**Pull request merged**